guides/Getting Started.md

```diff
@@ -0,0 +1,696 @@
+
+On the final line, rather than inserting the `person`, we insert the `changeset`. The `changeset` knows about the `person`, the changes and the validation rules that must be met before the data can be entered into the database. When this third line runs, we'll see this:
+
+```elixir
+{:error,
+ #Ecto.Changeset<action: :insert, changes: %{},
+  errors: [first_name: "can't be blank", last_name: "can't be blank"],
+  data: #Friends.Person<>, valid?: false>}
+```
+
+Just like the last time we did an insert, this returns a tuple. This time however, the first element in the tuple is `:error`, which indicates something bad happened. The specifics of what happend are included in the changeset which is returned. We can access these by doing some pattern matching:
+
+```elixir
+{ :error, changeset } = Friends.Repo.insert changeset
+```
+
+Then we can get to the errors by doing `changeset.errors`:
+
+```elixir
+[first_name: "can't be blank", last_name: "can't be blank"]
+```
+
+And we can ask the changeset itself it is valid, even before doing an insert:
+
+```
+changeset.valid?
+false
+```
+
+Since this changeset has errors, no new record was inserted into the `people`
+table.
+
+Let's try now with some valid data.
+
+```elixir
+person = %Friends.Person{}
+changeset = Friends.Person.changeset(person, %{first_name: "Ryan", last_name: "Bigg"})
+```
+
+We start out here with a normal `Friends.Person` struct. We then create a changeset for that `person` which has a `first_name` and a `last_name` parameter specified. At this point, we can ask the changeset if it has errors:
+
+```elixir
+changeset.errors
+[]
+```
+
+And we can ask if it's valid or not:
+
+```elixir
+changeset.valid?
+true
+```
+
+The changeset does not have errors, and is valid. Therefore if we try to insert this changeset it will work:
+
+```elixir
+Repo.insert changeset
+
+{:ok,
+ %Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: nil,
+  first_name: "Ryan", id: 3, last_name: "Bigg"}}
+```
+
+
+Due to `Friends.Repo.insert` returning a tuple, we can use a `case` to determine different code paths depending on what happens:
+
+```elixir
+case Friends.Repo.insert(changeset) do
+  { :ok, person } ->
+    # do something with person
+  { :error, changeset } ->
+    # do something with changeset
+end
+```
+
+**NOTE:** `changeset.valid?` will not check constraints (such as `uniqueness_constraint`). For that, you will need to attempt to do an insert and check for errors. It's for this reason it's best practice to try inserting data and validation the returned tuple from `Friends.Repo.insert` to get the correct errors, rather than relying on the changeset itself.
+
+If the insertion of the changeset succeeds, then you can do whatever you wish with the `person` returned in that result. If it fails, then you have access to the changeset and its errors. In the failure case, you may wish to present these errors to the end user.
```

Comment:
We could mention the `Ecto.Changeset.traverse_errors/2` function here that allows to easily format errors to show to the end user.